script/unicode/Verification.lean

```diff
@@ -0,0 +1,49 @@
+/-
+Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jean-Baptiste Tristan
+-/
+import Init.Data.Char.UnicodeSkipList
+import Init.Data.Char.Tables
+import Unicode.Reference
+import Unicode.Parse
+
+open System IO FilePath Process FS Std Char.UnicodeSkipList
+
+def compareTables (ucd : List UnicodeData) (property : UnicodeData → Bool) : IO Bool := do
+  let time ← monoMsNow
+  let mut failed := false
+  let table := numericTable
+  let referenceTable := referenceTable ucd property
+  for i in Range.mk 0 1114112 1 do
+    let c := Char.ofNat i
+    let ref := referenceSearch referenceTable c
+    let candidate := search table c
+    if ref ≠ candidate then
+      failed := true
+      println s!"{c.toNat} {c} {ref} {candidate}"
```

Comment:
I think it would be good to provide a slighly more useful error message here, so that it's easier to tell at a glance what the problem is if CI fails.